weblate/trans/forms.py

```diff
@@ -133,6 +133,40 @@ def to_python(self, value):
         return value
 
 
+class DateRangeInput(forms.TextInput):
+    """Input compatible with date range picker."""
+
+    template_name = "snippets/range-field.html"
+
+
+class DateRangeField(forms.CharField):
+    """Field for a date range input."""
+
+    def __init__(self, **kwargs) -> None:
+        if "widget" not in kwargs:
+            kwargs["widget"] = DateRangeInput
+        super().__init__(**kwargs)
+
+    def to_python(self, value):
+        start, end = value.split(" - ")
+        return {
+            "start_date": from_current_timezone(datetime.strptime(start, "%m/%d/%Y")),
+            "end_date": from_current_timezone(datetime.strptime(end, "%m/%d/%Y")),
+        }
```

Comment:
This will crash on invalid input instead of providing user a validation error.

Reply:
To solve this, I added a exception to handle it in case the input is invalid.